### PR TITLE
Display git tags instead of short hash when in deteached head state.

### PR DESCRIPTION
**Pull request by rickard-von-essen**
Use `git describe --tags` to get a short description of the commit
relative the nearest tag. Only the tag name will be shown if your
currently on a tag.

### CODE REVIEW

**Comment by rickard-von-essen** [comment]
Example screenshot
screen shot 2016-08-21 at 21 21 29

File: fish_prompt.fish [hunk 1]
```diff
@@ -53,7 +54,12 @@
 function __bobthefish_git_branch -S -d 'Get the current git branch (or commitish)'
   set -l ref (command git symbolic-ref HEAD ^/dev/null)
   if [ $status -gt 0 ]
-    set -l branch (command git show-ref --head -s --abbrev | head -n1 ^/dev/null)
+    set -l branch
+    if [ "$theme_display_git_tag" = 'yes' ]
+      set branch (command git describe --tags ^/dev/null)
```
Comment (bobthecow) [inline fish_prompt.fish]: this probably needs an `--always` in there to keep it from falling over if you don't have a tag in your repo yet.

**Comment by bobthecow** [comment]
I've been considering changing the detached head "branch name" command to this:

```
command git describe --tags --exact-match ^/dev/null;
  or command git show-ref --head -s --abbrev | head -n1 ^/dev/null
```

Which would be very similar to yours, except it would show only the hash when HEAD isn't a tag itself, while yours would do something like `v0.1.4-1-g4f7d5e7`. Would having the "tag-or-hash but not both" version be enough for you, or are you actually looking for the extended "tag-plus-hash-plus-more" information?

**Comment by rickard-von-essen** [comment]
That would be great. Tags are most important.

**Comment by bobthecow** [comment]
This is fixed in 4669e3dad6ad

Let me know if you have any issues :)

**Comment by rickard-von-essen** [comment]
Works nicely!

**Pull request closed without merging**